### ISSUE TITLE
Question about the loss backward with RWKV7

### ISSUE THREAD

**Issue by Cynicarlos** [BlinkDL/RWKV-LM#309]
Hi, when I try to apply rwkv7 to vision task by using Block as the core module, I found the output consist some NaN value, which made the loss can not backward. I also found that the input would have NaN value just pass one block. I'm sure the original input data is no problem. I wonder what the possible cause is.

**Comment by BlinkDL** [comment]
Hi got code? :)

**Comment by Cynicarlos** [comment]
An example as follows, the random input passing through just one block will have nan value.
```python
class Block(nn.Module):
    def __init__(self, dims, head_size, layer_id):
        super().__init__()
        self.layer_id = layer_id

        self.ln0 = nn.LayerNorm(dims)
        self.ln1 = nn.LayerNorm(dims)
        self.ln2 = nn.LayerNorm(dims)

        self.att = RWKV_Tmix_x070(dims=dims, head_size_a=head_size, layer_id=layer_id)
        self.ffn = RWKV_CMix_x070(dims=dims, layer_id=layer_id)

    def forward(self, x, v_first):
        b, c, h, w = x.shape
        x = rearrange(x, 'b c h w -> b (h w) c')
        v_first = rearrange(v_first, 'b c h w -> b (h w) c')

        if self.layer_id == 0:
            x = self.ln0(x)
        xx, v_first = self.att(self.ln1(x), v_first)
        x = x + xx
        x = x + self.ffn(self.ln2(x))

        x = rearrange(x, 'b (h w) c -> b c h w', h=h, w=w)
        v_first = rearrange(v_first, 'b (h w) c -> b c h w', h=h, w=w)

        return x, v_first


if __name__ == '__main__':
    model = Block(dims=32, head_size=32, layer_id=0).to(device='cuda') 
    x = torch.randn((1,32,384,384)).to(device='cuda')
    v_first = torch.empty_like(x)
    y, _ = model(x, v_first)
    print(torch.isnan(y).any())
```

**Comment by Cynicarlos** [comment]
Hi, it seems that the problem has something to do with the ''dims'', I found that there would be Nan value only if the dims is less than 128. And when I set the dims to 128 or bigger,  it's ok to output normal value. That is interesting, any ideas?

**Comment by BlinkDL** [comment]
Please firstly replicate the loss here:
https://github.com/BlinkDL/RWKV-LM/tree/main/RWKV-v7/train_temp

Then gradually change it to use your model size & data & setup :)
